formatter_test.go

```diff
@@ -587,6 +587,120 @@ func TestFormatter_FloatFormat(t *testing.T) {
 	}
 }
 
+func TestFormatter_DigitSeparator(t *testing.T) {
```

Comment:
We also need to cover a few more cases:

* big integer part, small decimal part, non scientific 
* big integer part, small decimal part, scientific 
* big integer part, big decimal part, non scientific 
* big integer part, big decimal part, scientific 

By small decimal part, I mean 1 to 3 digits, by big - more than 3 digits.